### PR TITLE
small fix secret autoloading, bump azure

### PR DESCRIPTION
**Pull request by samansmink**
fixes:
- autoloading errors a bit nicer
- bump azure
- azure now autoloads correctly for all its prefixes

### CODE REVIEW

**Comment by Mytherin** [comment]
Thanks! Could you just have a look at the CI failures?

**Comment by samansmink** [comment]
Just disabled the statement that failed for now. 

Properly testing autoloading out-of-tree extensions will need some extra work. Currently we do this sortof "best-effort" where duckdb just runs all out-of-tree extension tests in autoload mode, but because none of the azure tests are actually runnable without azure test servers, none of them run.

File: src/include/duckdb/main/extension_entries.hpp [hunk 1]
```diff
@@ -311,10 +311,9 @@ static constexpr ExtensionEntry EXTENSION_COLLATIONS[] = {
 // Note: these are currently hardcoded in scripts/generate_extensions_function.py
 // TODO: automate by passing though to script via duckdb
 static constexpr ExtensionEntry EXTENSION_FILE_PREFIXES[] = {
-    {"http://", "httpfs"}, {"https://", "httpfs"}, {"s3://", "httpfs"}, {"s3a://", "httpfs"}, {"s3n://", "httpfs"},
-    {"gcs://", "httpfs"},  {"gs://", "httpfs"},    {"r2://", "httpfs"}
-    //    {"azure://", "azure"}
-}; // END_OF_EXTENSION_FILE_PREFIXES
+    {"http://", "httpfs"}, {"https://", "httpfs"}, {"s3://", "httpfs"},  {"s3a://", "httpfs"},
+    {"s3n://", "httpfs"},  {"gcs://", "httpfs"},   {"gs://", "httpfs"},  {"r2://", "httpfs"},
++    {"azure://", "azure"}, {"az://", "azure"},     {"abfss://", "azure"}}; // END_OF_EXTENSION_FILE_PREFIXES
```
Comment (carlopi) [inline src/include/duckdb/main/extension_entries.hpp:316]:
```suggestion
    {"azure://", "azure"}, {"az://", "azure"},     {"abfss://", "azure"}}; // END_OF_EXTENSION_FILE_PREFIXES
```

Comment (Mytherin) [reply src/include/duckdb/main/extension_entries.hpp]: Thanks - the web editor is not so great sometimes :)

**Comment by carlopi** [comment]
I think this is good to go, only minor nit we have also the extensions listed in the csv, so maybe also there it should be updated. Unsure if that has any use, probably should just be checked and removed.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**